SimTracker/Common/src/SiG4UniversalFluctuation.cc

```diff
@@ -222,7 +207,7 @@ double SiG4UniversalFluctuation::SampleFluctuations(const double momentum,
   //
   if (suma > sumalim)
   {
-    p1 = 0., p2 = 0 ;
+    double p1, p2;
```

Comment:
HI David, the full block:
double p1, p2;
    if((a1+a2) > 0.)
    {
      // excitation type 1
      if (a1>alim) {
        siga=sqrt(a1) ;
        p1 = max(0., CLHEP::RandGaussQ::shoot(engine, a1, siga) + 0.5);
      } else {
	p1 = double(CLHEP::RandPoissonQ::shoot(engine,a1));
      }
    
      // excitation type 2
      if (a2>alim) {
        siga=sqrt(a2) ;
        p2 = max(0., CLHEP::RandGaussQ::shoot(engine, a2, siga) + 0.5);
      } else {
	p2 = double(CLHEP::RandPoissonQ::shoot(engine,a2));
      }
    
      loss = p1*e1Fluct+p2*e2Fluct;
 
      // smearing to avoid unphysical peaks
      if (p2 > 0.)
        loss += (1.-2.*CLHEP::RandFlat::shoot(engine))*e2Fluct;
      else if (loss>0.)
        loss += (1.-2.*CLHEP::RandFlat::shoot(engine))*e1Fluct;
      if (loss < 0.) loss = 0.0;
    }

so, for both p1 and p2 we have if()p1=value1; else p1=value2, the same for p2 - both are defined in any case.

However, what I see now: the line 'double p1, p2;' may be moved down inside the block. Minor thing, should I do this?